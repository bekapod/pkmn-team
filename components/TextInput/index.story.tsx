// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import TextInput from ".";

storiesOf("TextInput", module).add("default", () => <TextInput type="text" />);
