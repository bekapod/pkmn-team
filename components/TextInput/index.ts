import { placeholder } from "polished";
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";

interface Props {
  isInvalid?: boolean;
}

const TextInput = styled.input`
  width: 300px;
  padding: 0 ${variables.spacing.sm}px;
  font-family: ${variables.fonts.base};
  font-size: ${variables.fontSizes.base}px;
  font-weight: 900;
  line-height: ${lineHeight("base")};
  background-color: ${variables.colors.white};
  box-shadow: ${({ isInvalid }: Props) =>
    isInvalid
      ? `0 0 0 ${variables.spacing.xs}px ${variables.colors.error}`
      : "none"};
  border: none;

  &:focus {
    outline: none;
    box-shadow: 0 0 0 ${variables.spacing.xs}px ${variables.colors.secondary};
  }

  ${placeholder({
    color: variables.colors.gray,
    "text-transform": "uppercase"
  })};
`;

export default TextInput;
