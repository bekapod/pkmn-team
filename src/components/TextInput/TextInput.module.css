.base {
  width: 300px;
  padding: 0 var(--spacing-3);
  font-family: var(--font-family);
  font-size: var(--font-size-base);
  line-height: var(--line-height-xl);
  font-weight: 700;
  background-color: var(--color-white);
  border: none;

  &[aria-invalid] {
    box-shadow: 0 0 0 var(--spacing-1) var(--color-error);
  }

  &:focus {
    outline: none;
    box-shadow: 0 0 0 var(--spacing-1) var(--color-downy-600);
  }

  &::placeholder {
    color: var(--color-sugar-cane-700);
    font-weight: 900;
    text-transform: uppercase;
  }
}
