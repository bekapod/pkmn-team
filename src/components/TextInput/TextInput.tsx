import { ComponentPropsWithRef, forwardRef, FunctionComponent } from 'react';
import cx from 'classnames';

export type TextInputProps = ComponentPropsWithRef<'input'> & {
  isInvalid?: boolean;
};

export const TextInput: FunctionComponent<TextInputProps> = forwardRef(
  ({ isInvalid, className, ...props }, ref) => (
    <input
      ref={ref}
      type="text"
      className={cx('input-text', className)}
      aria-invalid={isInvalid ? true : undefined}
      {...props}
    />
  )
);
