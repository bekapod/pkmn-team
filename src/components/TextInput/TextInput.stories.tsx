import { Meta, Story } from '@storybook/react/types-6-0';
import { ComponentProps } from 'react';
import { TextInput, TextInputProps } from './TextInput';

export default {
  title: 'Components/Text Input',
  component: TextInput,
  args: {
    disabled: false
  }
} as Meta<TextInputProps>;

export const textInput: Story<ComponentProps<typeof TextInput>> = args => (
  <TextInput {...args} />
);

export const withPlaceholder: Story<ComponentProps<typeof TextInput>> =
  args => <TextInput placeholder="Some placeholder text" {...args} />;
