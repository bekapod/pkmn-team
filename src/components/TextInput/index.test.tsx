import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
import TextInput from "./";

describe("<TextInput />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(
      <TextInput type="text" placeholder="Text Input" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders without crashing when invalid", () => {
    const tree = renderer.create(
      <TextInput type="text" placeholder="Text Input" isInvalid={true} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
