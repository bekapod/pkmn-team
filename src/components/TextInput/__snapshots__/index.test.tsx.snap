// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders without crashing 1`] = `
<input
  className="TextInput-sc-149814h-0 kOVbBA"
  placeholder="Text Input"
  type="text"
/>
`;

exports[`<TextInput /> renders without crashing when invalid 1`] = `
<input
  className="TextInput-sc-149814h-0 gghhmx"
  placeholder="Text Input"
  type="text"
/>
`;
